docs/newtypes/index.md

```diff
@@ -0,0 +1,310 @@
+```
+
+This can be particularly useful because since the representations of the new type and the underlying types are the same at runtime, operators that are not strongly typed like `println` and `==` will not respect the difference between the new type and the underlying type. Using the functional abstractions in ZIO Prelude we can avoid this problem.
+
+## Subtypes
+
+When we use `Newtype` the new type we are creating is completely distinct from the underlying type. This can be useful when we want to "hide" the underlying representation of the new type except when we explicitly unwrap it.
+
+However, it can lead to boilerplate in some cases where we need to explicitly unwrap the new type or reimplement functionality that already exists on the underlying type.
+
+For example, say we want to test whether one `SequenceNumber` is after another. Right now this will not work:
+
+```scala mdoc
+// SequenceNumber(2) > SequenceNumber(1) // does not compile
+```
+
+The `>` operator is defined on `Int` and as far as the Scala compiler is concerned `SequenceNumber` and `Int` are completely unrelated types. Of course we could unwrap each of our sequence numbers or define a new `>` operator on sequence numbers but here we are not taking advantage of the fact that we know that every `SequenceNumber` is an `Int`.
+
+We can fix this by extending `Subtype` instead of `Newtype` when creating our new type.
+
+```scala mdoc:reset-object
+import zio.prelude.Subtype
+
+object SequenceNumber extends Subtype[Int]
+type SequenceNumber = SequenceNumber.Type
+
+SequenceNumber(2) > SequenceNumber(1) // okay
+```
+
+Now `SequenceNumber` is a type that is different from `Int` but is still a subtype of `Int`.
+
+This means that we can use a `SequenceNumber` any time we need an `Int` and can use operators defined on `Int` on `SequenceNumber`. However, we still get the type safety os not being able to use an `Int` or `AccountNumber` when a `SequenceNumber` is expected.
+
+## Smart Types
+
+So far, all the new types we have created have been distinct from the underlying types but have not imposed any additional constraints on the values that the underlying type can take. For example, the `SequenceNumber` above could in principle be any `Int` value, whether that is `42` or `-1`.
+
+In many cases that makes sense. We often want to use a new type to denote that this particular `Int` value represents a sequence number and shouldn't be confused with another `Int` value that represents an account number without imposing other constraints.
+
+However, in other cases we may want to restrict the values that the underlying type can take. For instance, perhaps a `SequenceNumber` should never be negative.
+
+We can model this in ZIO Prelude by extending `NewtypeSmart` and `SubtypeSmart` when creating our new type and providing an `Assertion` that describes the constraints on the underlying value. We call these types smart types because they build in the logic for their own validation.
```

Comment:
you could say 1-2 lines why you don't give examples for `NewtypeSmart` and/or why they're less interesting.

Reply:
Added.